### DISCUSSION TITLE
Should "Introduce a local variable" work with top-level statements?

### DISCUSSION THREAD

**Discussion by BhaaLseN** [dotnet/roslyn#67267] [Q&A] [answered]
I just noticed that [Introduce a local variable](https://learn.microsoft.com/en-us/visualstudio/ide/reference/introduce-local-variable) doesn't show up for me; and I've always made heavy use of it.

The only thing that differs (which I just confirmed) is that this is a one-off program where I use [top-level statements](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/top-level-statements) for simplicity.
![top-level statements](https://user-images.githubusercontent.com/826052/224481363-4edee97e-3314-45fe-9359-b86afc7baabf.png)

As soon as I convert this to a Main-style program, the refactoring appears again.
![Program.Main style](https://user-images.githubusercontent.com/826052/224481410-4032a899-55a2-443e-ba05-2eaac3c7f897.png)

Is this expected, or just an oversight in the code fix provider?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Just an oversight. Wood happily take a contribution here:-)
